tests/devapps/MSI/HelperService/MSIHelperService/Helper/MSIHelper.cs

```diff
@@ -0,0 +1,370 @@
+            Status500InternalServerError = StatusCodes.Status500InternalServerError,
+            Status503ServiceUnavailable = StatusCodes.Status503ServiceUnavailable
+        }
+
+        //Azure Resources
+        internal enum AzureResource
+        {
+            webapp,
+            function,
+            vm,
+            azurearc,
+            servicefabric,
+            cloudshell
+        }
+
+        /// <summary>
+        /// Gets the Environment Variable from the Azure Web App
+        /// </summary>
+        /// <param name="variableName"></param>
+        /// <param name="logger"></param>
+        /// <returns>Returns the environment variable</returns>
+        public static string? GetWebAppEnvironmentVariable(
+            string variableName,
+            ILogger logger)
+        {
+            //On the Web app we have stored some environment variables, those variables 
+            //hold values that are used to call into several other azure resources
+            //Hide the App ID and App Secret and other variableds in this API
+            //these are used to get token for other azure test resources and stored in the
+            //web apps environment variables
+
+            logger.LogInformation("GetWebAppEnvironmentVariable Function called.");
+
+            return
+                s_protectedEnvironmentVariables.Any(s => s.Contains(variableName, StringComparison.OrdinalIgnoreCase)) ?
+                null :
+                Environment.GetEnvironmentVariable(variableName);
+        }
+
+        /// <summary>
+        /// Gets the Environment Variables from the Azure Web App
+        /// </summary>
+        /// <param name="variableName"></param>
+        /// <param name="logger"></param>
+        /// <returns>Returns the environment variable</returns>
+        public static Dictionary<string, string> GetWebAppEnvironmentVariables(
+            ILogger logger)
+        {
+            //Gets Azure Web App Specific environment variables and sends it back
+            //Sending back the specific ones that is needed for the MSI tests
+            Dictionary<string, string>? keyValuePairs = new();
+            keyValuePairs.Add("IDENTITY_HEADER", Environment.GetEnvironmentVariable("IDENTITY_HEADER") ?? "");
+            keyValuePairs.Add("IDENTITY_ENDPOINT", Environment.GetEnvironmentVariable("IDENTITY_ENDPOINT") ?? "");
+            keyValuePairs.Add("IDENTITY_API_VERSION", Environment.GetEnvironmentVariable("IDENTITY_API_VERSION") ?? "");
+
+            logger.LogInformation("GetWebAppEnvironmentVariables Function called.");
+
+            return keyValuePairs;
+        }
+
+        /// <summary>
+        /// Gets the Environment Variable from the Azure Function
+        /// </summary>
+        /// <param name="variableName"></param>
+        /// <param name="logger"></param>
+        /// <returns></returns>
+        public static string? GetFunctionAppEnvironmentVariable(
```

Comment:
Why don't you make these methods async to avoid Task.Run ?

Reply:
Generally it's better to have everything async. Hiding async is bad practice, it's inneficient (you spin an extra thread in that Task.Run). In some languages like NodeJS you can't even do sync over async - there's always a way to expose async :)